Slider/Assets/Scripts/Map/Jungle/JungleSignAnimator.cs

```diff
@@ -91,4 +105,27 @@ public void UpdateDirection()
             SetRandomDirection();
         }
     }
+
+    public void UpdateShape()
+    {
+        Hut h = null;
+
+        GameObject parent = this.transform.parent.gameObject;
+
+        if (parent != null)
+        {
+            GameObject grandparent = parent.transform.parent.gameObject;
+            if (grandparent != null)
+            {
+                h = grandparent.GetComponent<Hut>();
```

Comment:
do u think u can make this and the `UpdateDirection()` methods not use `parent.transform.gameObject` and just have a reference to `Hut` or `Sign` bc traversing the unity heirarchy is really bad o 3 o